### PR TITLE
(Feature) Search tokens (in Mainnet)

### PR DESCRIPTION
**Pull request by vbaranov**
Relates to https://github.com/poanetwork/metamask-extension/issues/113
Relates to https://github.com/poanetwork/metamask-extension/issues/134

The first step of tokens management support in NW.
The ability to search tokens by name, symbol, contract address. The tokens are from Mainnet only, as for now.

Search screen:

Confirm screen:

@dennis00010011b could ypu please update e2e tests to support this feature?

### CODE REVIEW

**Comment by dennis00010011b** [comment]
@vbaranov 
Add tokens-Search:
 button 'Next' always active if it was once activated
screen shot 2018-09-14 at 9 30 37 am

**Comment by dennis00010011b** [comment]
@vbaranov 
Add tokens-Search-Add token:
 app is cropped in full screen mode
screen shot 2018-09-14 at 9 20 47 am

**Comment by dennis00010011b** [comment]
@vbaranov 
>The tokens are from Mainnet only, as for now.

Mainnet’s tokens is searchable with any provider 
screen shot 2018-09-15 at 3 31 19 am

**Comment by dennis00010011b** [comment]
@vbaranov 
Long tokens name isn't fully displayed  and there is no option to spread it out

screen shot 2018-09-15 at 4 13 13 am

**Comment by dennis00010011b** [comment]
@vbaranov 
Main screen: list of added tokens is not scrollable

screen shot 2018-09-15 at 4 27 01 am
screen shot 2018-09-15 at 4 25 35 am

**Comment by dennis00010011b** [comment]
@vbaranov 
Add token dialog: typo, should be `this token` if one token in the list

screen shot 2018-09-15 at 7 27 45 am

**Comment by dennis00010011b** [comment]
@vbaranov 
Add token dialog: it would be helpful to have option to delete some token/tokens from the list before addition

screen shot 2018-09-15 at 7 34 53 am

**Comment by vbaranov** [comment]
@dennis00010011b 
>app is cropped in full screen mode

Fixed

>Mainnet’s tokens is searchable with any provider

I've hidden search tab for networks, other than Mainnet

>Long tokens name isn't fully displayed and there is no option to spread it out

Tooltip is added, if a token has long name

>Add token dialog: typo, should be this token if one token in the list

It is not a typo, but now text in the dialog depends on the number of tokens

>Main screen: list of added tokens has no scrollbar
After scrolling down the main menu is invisible

and

>Add token dialog: it would be helpful to have option to delete some token/tokens from the list before addition

Please create feature requests. I suggest to do it in a separate PRs.

**Comment by dennis00010011b** [comment]
@vbaranov 
Confirm list is cleared if user goes back from confirm screen to main screen. 
It can be done by clicking  'Cancel' on search screen or close and open wallet.  Is it intentional behavior?

**Comment by vbaranov** [comment]
yes, @dennis00010011b this is intentional: pending tokens are cleared on `cancel` or unmounting of view.

In frames of this issue, we also need to fix integration tests. I've added related issue ☝️. Please take a look.

**Comment by vbaranov** [comment]
Hi @fvictorio @patitonar - finally, it is ready for review

**Comment by fvictorio** [comment]
Can the "searchResults" label be changed to "Search results:"?

image

**Comment by fvictorio** [comment]
And maybe "Would you like to add this token?" should be "Would you like to add these tokens?".

File: old-ui/app/components/add-token/token-list/token-list.component.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+          <div className="token-list__tokens-container">
+            {
+              Array(6).fill(undefined)
+                .map((_, i) => {
+                  const { logo, symbol, name, address } = results[i] || {}
+                  const tokenAlreadyAdded = checkExistingAddresses(address, tokens)
+                  const title = `${name} (${symbol})`
+                  const isLongTitle = title.length > 28
+
+                  const tokenRow = (key) => h(`.${classnames('token-list__token', {
```
Comment (patitonar) [inline old-ui/app/components/add-token/token-list/token-list.component.js]: I think we should use JSX here too to be consistent with the rest of the file

Comment (vbaranov) [reply old-ui/app/components/add-token/token-list/token-list.component.js]: Thank you, agree. react-hyperscript was changed to JSX.

**Comment by vbaranov** [comment]
>Can the "searchResults" label be changed to "Search results:"?

Just removed the text. It takes some space but has no great meaning.

>And maybe "Would you like to add this token?" should be "Would you like to add these tokens?".

If there are multiple instances of tokens are chosen, you'll see "Would you like to add these tokens?"

>These files seem to have been added by mistake:

Yes, removed them from PR.

Thank you @fvictorio @patitonar for valuable comments. They were processed. I returned back this PR to reviewal stage.

**Pull request merged**